File: Week_06/java/src/_122_BestTimeToBuyAndSellStockII.java
class _122_BestTimeToBuyAndSellStockII {
    /**
     * 贪心算法
     * @param prices
     * @return
     */
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length < 2) {
            return 0;
        } 
        
        int ret = 0;
        int n = prices.length;
        for (int i = 1; i < n; i++) {
            ret += Math.max(0, prices[i] - prices[i - 1]);
        }
        return ret;
    }
}

File: Week_04/cpp/src/_122_BestTimeToBuyAndSellStockII.cpp
#include <vector>
using namespace std;
class UsingGreedy
{
public:
    // O(n) O(1)
    int maxProfit(vector<int> &prices)
    {
        int ret = 0;
        int n = prices.size();
        if (!n)
            return ret;
        for (int i = 1; i < n; i++)
        {
            ret += max(0, prices[i] - prices[i - 1]);
        }
        return ret;
    }
};